### PR TITLE
 engine: Write `/run/ignition.json` atomically

### PR DESCRIPTION
**Pull request by cgwalters**
I just noticed this while reading the code; in practice, the write
should never fail, but we really want to write files atomically
as a general rule.  Particularly in this case as the code may
come back and try to read it again, and if we only did a partial
write we'd just continually fail to read it.

### CODE REVIEW

**Comment by cgwalters** [comment]
For the record I didn't try the tests yet, but I did build+run locally and verified via `sha256sum bin/amd64/ignition` matched the version extracted from the initramfs in my cosa build + `cosa run` which worked.

**Comment by ajeddeloh** [comment]
ok to test

File: internal/exec/engine.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ import (
 	"os"
 	"time"
 
+	"github.com/google/renameio"
```
Comment (arithx) [inline internal/exec/engine.go]: Very minor: should probably move this import down to go with the two other non-stdlib external imports (the `vcontext` imports)

Comment (cgwalters) [reply internal/exec/engine.go]: Done

**Comment by ajeddeloh** [comment]
gofmt is failing

**Pull request merged**